R-package/R/osf-interactions.R

```diff
@@ -0,0 +1,174 @@
+# Load utility functions
+source(file.path(getwd(), "R", "create-release-files.R"))
+
+#' Update OSF files
+#'
+#' This function updates files on the Open Science Framework (OSF) using the provided OSF link and token.
+#' It downloads the dictionary file, validates the dictionary version and files sheet, and stages the files to be uploaded to OSF.
+#'
+#' @param OSF_REPO_LINK The link to the OSF repo.
+#' @param OSF_TOKEN The OSF token used to authenticate the user.
+#' @param dictionary_path The path to the dictionary file. If NULL, the file will be downloaded from OSF.
+#' @param past_dictionary_path The path to the past dictionary file. If NULL, the file will be downloaded from OSF.
+#'
+update_osf <- function(OSF_REPO_LINK,
+                       OSF_TOKEN,
+                       dictionary_path = NULL,
+                       past_dictionary_path = NULL){
+  # Setup logging
+  # Remove previous log
+  file.remove(odm_dictionary$log_path)
+  # Set logger appending
+  logger::log_appender(logger::appender_file(odm_dictionary$log_path))
+  
+  # Download file using passed credentials
+  dictionary_path <- download_dictionary(dictionary_path, OSF_TOKEN, OSF_REPO_LINK, odm_dictionary$tmp_dictionary_directory, origin_directory = "dev-release")
+  
+  # Validate dictionary version
+  dictionary_info <- get_dictionary(dictionary_path)
+  
+  dictionary <- dictionary_info[[1]]
+  dictionary_version <- dictionary_info[[2]]
+  
+  # Validate files sheet
+  validation_return <-
+    validate_and_parse_files_sheet(dictionary,
+                                   dictionary_version)
+  files_to_make <- validation_return[[1]]
+  fatal_errors_flag <- validation_return[[2]]
+  
+  # Will move with development
+  # Will become stop once function development is finished
+  if (fatal_errors_flag) {
+    warning(
+      "Errors were detected further building cannot continue please check the log for additional info"
+    )
+  }
+  
+  stage_osf_files(files_to_make,
+               dictionary)
+
+  # Download previous release dictionary
+  past_dictionary_path <- download_dictionary(past_dictionary_path, OSF_TOKEN, OSF_REPO_LINK, odm_dictionary$tmp_dictionary_directory_past_release, origin_directory = "Current Release")
```

Comment:
This and the next lines which parses the dictionary should be pulled into a function that downloads and parses the currently released dictionary from osf. Would also be used in the `create_release_files` function.